src/installer/managed/Microsoft.NET.HostModel/ComHost/TypeLibReader.cs

```diff
@@ -0,0 +1,73 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+
+using System;
+using System.Buffers.Binary;
+
+namespace Microsoft.NET.HostModel.ComHost
+{
+    /// <summary>
+    /// Reads data from a COM Type Library file based on the official implementation.
+    /// </summary>
+    internal class TypeLibReader
+    {
+        private byte[] tlbBytes;
+
+        public TypeLibReader(byte[] tlbBytes)
+        {
+            this.tlbBytes = tlbBytes;
+        }
+
+        private const int OffsetOfGuidOffset = sizeof(int) * 2;
+        private const int SizeOfGuidOffset = sizeof(int);
+
+        private const int OffsetOfMajorVersion = OffsetOfGuidOffset + SizeOfGuidOffset + sizeof(int) * 2 + sizeof(ushort) * 2;
+        private const int SizeOfMajorVersion = sizeof(ushort);
+        private const int OffsetOfMinorVersion = OffsetOfMajorVersion + SizeOfMajorVersion;
+        private const int SizeOfMinorVersion = sizeof(ushort);
+
+        private const int OffsetOfTypeInfosCount = OffsetOfMinorVersion + SizeOfMinorVersion + sizeof(int);
+        private const int SizeOfTypeInfosCount = sizeof(int);
+
+        private const int OffsetOfTablesStart = OffsetOfTypeInfosCount + SizeOfTypeInfosCount + sizeof(int) * 12;
+        private const int NumTablesToSkip = 5;
+        private const int SizeOfTableHeader = sizeof(int) * 4;
```

Comment:
Is there public/linkable documentation for these offsets?

Reply:
To my knowledge there is not. I can link to the internal implementation this is based on if we want to.